renku_notebooks/util/jupyterhub_.py

```diff
@@ -37,18 +36,9 @@
 def make_server_name(namespace, project, branch, commit_sha):
     """Form a 16-digit hash server ID."""
     server_string = f"{namespace}{project}{branch}{commit_sha}"
-    return md5(server_string.encode()).hexdigest()[:16]
-
-
-def notebook_url(user, server_name, notebook=None):
-    """Form the notebook server URL."""
-    notebook_url = urljoin(
-        os.environ.get("JUPYTERHUB_BASE_URL"),
-        "user/{user[name]}/{server_name}/".format(user=user, server_name=server_name),
+    return "{project}-{hash}".format(
+        project=project[:46], hash=md5(server_string.encode()).hexdigest()[:16]
```

Comment:
we could even probably use only a 7-character hash and it would be more than enough - what do you think?

Reply:
right, I guess we can safely go down to 7 or 8 chars hash.
Not sure if going from the first 46 chars to 54 would be a noticeable difference for the user :slightly_smiling_face: